### ISSUE TITLE
Annotation targets should be taken in account by type system 

### ISSUE THREAD

**Issue by ddenisenko** [raml-org/raml-js-parser-2#501]
Currently following RAML 
```
#%RAML 1.0
title: test
annotationTypes:
  xz:
    allowedTargets: Method

/g:
  get:
    body:
      application/json:
         (xz): z
```
passes validation. This happens because type system did not looks in possible annotation targets while it validates types.

**Comment by ddenisenko** [comment]
Duplicating https://github.com/raml-org/typesystem-ts/issues/40

**Comment by KonstantinSviridov** [comment]
to be released in 1.1.5
